File: lookforbest_app/lib/features/robots/domain/robot.dart
/// 机器人数据模型（对应 API /robots/{id} 响应）
class Robot {
  final int id;
  final String name;
  final String? nameEn;
  final String modelNumber;
  final String slug;
  final String? description;
  final String? coverImageUrl;
  final String? model3dUrl;
  final bool has3dModel;
  final bool hasVideo;
  final RobotManufacturer manufacturer;
  final RobotCategory category;
  final RobotSpecs specs;
  final String? priceRange;
  final String status;
  final int viewCount;
  final int favoriteCount;
  final bool isFavorited;
  final List<RobotMedia> images;
  final List<RobotMedia> videos;

  const Robot({
    required this.id,
    required this.name,
    this.nameEn,
    required this.modelNumber,
    required this.slug,
    this.description,
    this.coverImageUrl,
    this.model3dUrl,
    required this.has3dModel,
    required this.hasVideo,
    required this.manufacturer,
    required this.category,
    required this.specs,
    this.priceRange,
    required this.status,
    required this.viewCount,
    required this.favoriteCount,
    required this.isFavorited,
    required this.images,
    required this.videos,
  });

  factory Robot.fromJson(Map<String, dynamic> json) {
    return Robot(
      id: json['id'] as int,
      name: json['name'] as String,
      nameEn: json['nameEn'] as String?,
      modelNumber: json['modelNumber'] as String,
      slug: json['slug'] as String,
      description: json['description'] as String?,
      coverImageUrl: json['coverImageUrl'] as String?,
      model3dUrl: json['model3dUrl'] as String?,
      has3dModel: json['has3dModel'] as bool? ?? false,
      hasVideo: json['hasVideo'] as bool? ?? false,
      manufacturer: RobotManufacturer.fromJson(
        json['manufacturer'] as Map<String, dynamic>,
      ),
      category: RobotCategory.fromJson(
        json['category'] as Map<String, dynamic>,
      ),
      specs: RobotSpecs.fromJson(json['specs'] as Map<String, dynamic>),
      priceRange: json['priceRange'] as String?,
      status: json['status'] as String? ?? 'active',
      viewCount: json['viewCount'] as int? ?? 0,
      favoriteCount: json['favoriteCount'] as int? ?? 0,
      isFavorited: json['isFavorited'] as bool? ?? false,
      images: (json['images'] as List<dynamic>?)
              ?.map((e) => RobotMedia.fromJson(e as Map<String, dynamic>))
              .toList() ??
          [],
      videos: (json['videos'] as List<dynamic>?)
              ?.map((e) => RobotMedia.fromJson(e as Map<String, dynamic>))
              .toList() ??
          [],
    );
  }
}

/// 机器人列表卡片（精简版）
class RobotSummary {
  final int id;
  final String name;
  final String slug;
  final String? coverImageUrl;
  final RobotManufacturer manufacturer;
  final RobotCategory category;
  final double? payloadKg;
  final int? reachMm;
  final int? dof;
  final bool has3dModel;
  final String? priceRange;
  final int viewCount;
  final int favoriteCount;

  const RobotSummary({
    required this.id,
    required this.name,
    required this.slug,
    this.coverImageUrl,
    required this.manufacturer,
    required this.category,
    this.payloadKg,
    this.reachMm,
    this.dof,
    required this.has3dModel,
    this.priceRange,
    required this.viewCount,
    required this.favoriteCount,
  });

  factory RobotSummary.fromJson(Map<String, dynamic> json) {
    return RobotSummary(
      id: json['id'] as int,
      name: json['name'] as String,
      slug: json['slug'] as String,
      coverImageUrl: json['coverImageUrl'] as String?,
      manufacturer: RobotManufacturer.fromJson(
        json['manufacturer'] as Map<String, dynamic>,
      ),
      category: RobotCategory.fromJson(
        json['category'] as Map<String, dynamic>,
      ),
      payloadKg: (json['payloadKg'] as num?)?.toDouble(),
      reachMm: json['reachMm'] as int?,
      dof: json['dof'] as int?,
      has3dModel: json['has3dModel'] as bool? ?? false,
      priceRange: json['priceRange'] as String?,
      viewCount: json['viewCount'] as int? ?? 0,
      favoriteCount: json['favoriteCount'] as int? ?? 0,
    );
  }
}

class RobotManufacturer {
  final int id;
  final String name;
  final String? logoUrl;
  final String? country;

  const RobotManufacturer({
    required this.id,
    required this.name,
    this.logoUrl,
    this.country,
  });

  factory RobotManufacturer.fromJson(Map<String, dynamic> json) {
    return RobotManufacturer(
      id: json['id'] as int,
      name: json['name'] as String,
      logoUrl: json['logoUrl'] as String?,
      country: json['country'] as String?,
    );
  }
}

class RobotCategory {
  final int id;
  final String name;
  final String? slug;

  const RobotCategory({required this.id, required this.name, this.slug});

  factory RobotCategory.fromJson(Map<String, dynamic> json) {
    return RobotCategory(
      id: json['id'] as int,
      name: json['name'] as String,
      slug: json['slug'] as String?,
    );
  }
}

class RobotSpecs {
  final double? payloadKg;
  final int? reachMm;
  final int? dof;
  final double? repeatabilityMm;
  final double? maxSpeedDegS;
  final double? weightKg;
  final String? ipRating;
  final String? mounting;

  const RobotSpecs({
    this.payloadKg,
    this.reachMm,
    this.dof,
    this.repeatabilityMm,
    this.maxSpeedDegS,
    this.weightKg,
    this.ipRating,
    this.mounting,
  });

  factory RobotSpecs.fromJson(Map<String, dynamic> json) {
    return RobotSpecs(
      payloadKg: (json['payloadKg'] as num?)?.toDouble(),
      reachMm: json['reachMm'] as int?,
      dof: json['dof'] as int?,
      repeatabilityMm: (json['repeatabilityMm'] as num?)?.toDouble(),
      maxSpeedDegS: (json['maxSpeedDegS'] as num?)?.toDouble(),
      weightKg: (json['weightKg'] as num?)?.toDouble(),
      ipRating: json['ipRating'] as String?,
      mounting: json['mounting'] as String?,
    );
  }
}

class RobotMedia {
  final int id;
  final String url;
  final String? thumbnailUrl;
  final String? title;

  const RobotMedia({
    required this.id,
    required this.url,
    this.thumbnailUrl,
    this.title,
  });

  factory RobotMedia.fromJson(Map<String, dynamic> json) {
    return RobotMedia(
      id: json['id'] as int,
      url: json['url'] as String,
      thumbnailUrl: json['thumbnailUrl'] as String?,
      title: json['title'] as String?,
    );
  }
}

/// 机器人筛选参数
class RobotFilter {
  final String? q;
  final int? categoryId;
  final int? manufacturerId;
  final List<int>? domainIds;
  final double? payloadMin;
  final double? payloadMax;
  final int? reachMin;
  final int? reachMax;
  final int? dofMin;
  final int? dofMax;
  final bool? has3dModel;
  final String? priceRange;
  final String sort;
  final int page;
  final int size;

  const RobotFilter({
    this.q,
    this.categoryId,
    this.manufacturerId,
    this.domainIds,
    this.payloadMin,
    this.payloadMax,
    this.reachMin,
    this.reachMax,
    this.dofMin,
    this.dofMax,
    this.has3dModel,
    this.priceRange,
    this.sort = 'relevance',
    this.page = 0,
    this.size = 20,
  });

  Map<String, dynamic> toQueryParams() {
    return {
      if (q != null && q!.isNotEmpty) 'q': q,
      if (categoryId != null) 'categoryId': categoryId,
      if (manufacturerId != null) 'manufacturerId': manufacturerId,
      if (domainIds != null && domainIds!.isNotEmpty)
        'domainIds': domainIds!.join(','),
      if (payloadMin != null) 'payloadMin': payloadMin,
      if (payloadMax != null) 'payloadMax': payloadMax,
      if (reachMin != null) 'reachMin': reachMin,
      if (reachMax != null) 'reachMax': reachMax,
      if (dofMin != null) 'dofMin': dofMin,
      if (dofMax != null) 'dofMax': dofMax,
      if (has3dModel != null) 'has3dModel': has3dModel,
      if (priceRange != null) 'priceRange': priceRange,
      'sort': sort,
      'page': page,
      'size': size,
    };
  }

  RobotFilter copyWith({
    String? q,
    int? categoryId,
    int? manufacturerId,
    List<int>? domainIds,
    double? payloadMin,
    double? payloadMax,
    int? reachMin,
    int? reachMax,
    int? dofMin,
    int? dofMax,
    bool? has3dModel,
    String? priceRange,
    String? sort,
    int? page,
    int? size,
  }) {
    return RobotFilter(
      q: q ?? this.q,
      categoryId: categoryId ?? this.categoryId,
      manufacturerId: manufacturerId ?? this.manufacturerId,
      domainIds: domainIds ?? this.domainIds,
      payloadMin: payloadMin ?? this.payloadMin,
      payloadMax: payloadMax ?? this.payloadMax,
      reachMin: reachMin ?? this.reachMin,
      reachMax: reachMax ?? this.reachMax,
      dofMin: dofMin ?? this.dofMin,
      dofMax: dofMax ?? this.dofMax,
      has3dModel: has3dModel ?? this.has3dModel,
      priceRange: priceRange ?? this.priceRange,
      sort: sort ?? this.sort,
      page: page ?? this.page,
      size: size ?? this.size,
    );
  }
}

File: backend/src/main/java/com/lookforbest/entity/Robot.java
package com.lookforbest.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;

@Entity
@Table(name = "robots")
@Getter
@Setter
public class Robot {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "manufacturer_id", nullable = false)
    private Manufacturer manufacturer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = false)
    private RobotCategory category;

    @Column(nullable = false, length = 300)
    private String name;

    @Column(name = "name_en", length = 300)
    private String nameEn;

    @Column(length = 500)
    private String subtitle;

    @Column(name = "model_number", length = 200)
    private String modelNumber;

    @Column(unique = true, nullable = false, length = 300)
    private String slug;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "description_en", columnDefinition = "TEXT")
    private String descriptionEn;

    @Column(columnDefinition = "TEXT")
    private String introduction;

    @Column(name = "application_scenarios", columnDefinition = "TEXT")
    private String applicationScenarios;

    @Column(columnDefinition = "TEXT")
    private String advantages;

    @Column(name = "release_year")
    private Short releaseYear;

    @Enumerated(EnumType.STRING)
    @Column(length = 20)
    private RobotStatus status = RobotStatus.active;

    // 技术参数
    private Byte dof;

    @Column(name = "payload_kg", precision = 10, scale = 3)
    private BigDecimal payloadKg;

    @Column(name = "reach_mm")
    private Integer reachMm;

    @Column(name = "repeatability_mm", precision = 8, scale = 4)
    private BigDecimal repeatabilityMm;

    @Column(name = "max_speed_deg_s", precision = 10, scale = 2)
    private BigDecimal maxSpeedDegS;

    @Column(name = "weight_kg", precision = 10, scale = 2)
    private BigDecimal weightKg;

    @Column(name = "ip_rating", length = 20)
    private String ipRating;

    @Column(length = 100)
    private String mounting;

    @Column(name = "max_speed_m_s", precision = 6, scale = 3)
    private BigDecimal maxSpeedMs;

    @Column(name = "max_load_kg", precision = 10, scale = 2)
    private BigDecimal maxLoadKg;

    @Column(name = "battery_life_h", precision = 5, scale = 1)
    private BigDecimal batteryLifeH;

    @Column(name = "navigation_type", length = 200)
    private String navigationType;

    @Column(name = "height_mm")
    private Integer heightMm;

    @Column(name = "walking_speed_m_s", precision = 5, scale = 3)
    private BigDecimal walkingSpeedMs;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "extra_specs", columnDefinition = "json")
    private Map<String, Object> extraSpecs;

    @Column(name = "price_range", length = 100)
    private String priceRange = "inquiry";

    @Column(name = "price_usd_from", precision = 12, scale = 2)
    private BigDecimal priceUsdFrom;

    @Column(name = "cover_image_url", length = 500)
    private String coverImageUrl;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "content_images", columnDefinition = "json")
    private List<Map<String, String>> contentImages;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "video_urls", columnDefinition = "json")
    private List<String> videoUrls;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
        name = "robot_tag_mappings",
        joinColumns = @JoinColumn(name = "robot_id"),
        inverseJoinColumns = @JoinColumn(name = "tag_id")
    )
    private Set<RobotTag> tags = new HashSet<>();

    @Column(name = "is_open_source", length = 20)
    private String isOpenSource;

    @Column(name = "listed_date")
    private LocalDate listedDate;

    @Column(name = "model_3d_url", length = 500)
    private String model3dUrl;

    @Column(name = "model_3d_preview", length = 500)
    private String model3dPreview;

    @Column(name = "has_3d_model")
    private Boolean has3dModel = false;

    @Column(name = "has_video")
    private Boolean hasVideo = false;

    @Column(name = "view_count")
    private Long viewCount = 0L;

    @Column(name = "favorite_count")
    private Integer favoriteCount = 0;

    @Column(name = "compare_count")
    private Integer compareCount = 0;

    @Column(name = "is_featured")
    private Boolean isFeatured = false;

    @Column(name = "is_verified")
    private Boolean isVerified = false;

    @Column(name = "sort_order")
    private Integer sortOrder = 0;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
        name = "robot_application_domains",
        joinColumns = @JoinColumn(name = "robot_id"),
        inverseJoinColumns = @JoinColumn(name = "domain_id")
    )
    private Set<ApplicationDomain> applicationDomains = new HashSet<>();

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public enum RobotStatus { active, discontinued, upcoming }
}
